Engines/Wine/Engine/Versions/script.js

```diff
@@ -1,8 +1,122 @@
-/* exported LATEST_STABLE_VERSION */
-module.LATEST_STABLE_VERSION = "4.0.3";
-/* exported LATEST_DEVELOPMENT_VERSION */
-module.LATEST_DEVELOPMENT_VERSION = "4.21";
-/* exported LATEST_STAGING_VERSION */
-module.LATEST_STAGING_VERSION = "4.21";
-/* exported LATEST_DOS_SUPPORT_VERSION */
-module.LATEST_DOS_SUPPORT_VERSION = "4.0";
+const {cat, touch} = include("utils.functions.filesystem.files");
+const Downloader = include("utils.functions.net.download");
+const propertyReader = Bean("propertyReader");
+
+/**
+ * Sorts an array of Wine versions
+ * @param {array} versions The versions array
+ * @returns {void}
+ */
+module.sortVersions = function (versions) {
+    versions.sort((a, b) =>
+    {
+        const aVersionNumbers = Array.from(a.split('.')).map(item => Number(item));
+        const bVersionNumbers = Array.from(b.split('.')).map(item => Number(item));
```

Comment:
Maybe it makes sense to first check that the version strings `a` and `b` have the correct format, e.g. consist of three number parts split by dots (a regex could be used for this).

Afterwards we should be able to "simplify" the sorting code:

```javascript
const [aMajor, aMinor, aPatch] = Array.from(a.split('.')).map(item => Number(item));
const [bMajor, bMinor, bPatch] = Array.from(b.split('.')).map(item => Number(item));

if (aMajor != bMajor) {
   return aMajor - bMajor;
}

if (aMinor != bMinor) {
   return aMinor - bMinor;
}

return aPatch - bPatch;
```

Reply:
The string doesn't have to be three numbers split by dots. You can have "4.0" or "5.0.0-rc2".